File: java/ch13/src/main/java/com/thesoftwaregorilla/tdd/money/Sum.java
package com.thesoftwaregorilla.tdd.money;

public class Sum implements Expression {
    final Money augend;
    final Money addend;

    public Sum(Money augend, Money addend) {
        this.augend = augend;
        this.addend = addend;
    }

    public Money reduce(String to) {
        return new Money(augend.getAmount() + addend.getAmount(), to);
    }
}

File: cs/ch13/Money/Sum.cs
namespace TheSoftwareGorilla.TDD.Money;

public class Sum : Expression
{
    public Money Augend { get; }
    public Money Addend { get; }

    public Sum(Money augend, Money addend)
    {
        Augend = augend;
        Addend = addend;
    }

    public Money Reduce(string to)
    {
        int amount = Augend.Amount + Addend.Amount;
        return new Money(amount, to);
    }
}